cmd/nvidia-dra-plugin/sharing.go

```diff
@@ -101,17 +101,32 @@ func (t *TimeSlicingManager) SetTimeSlice(devices *PreparedDevices, config *nasc
 		return fmt.Errorf("setting a TimeSlice duration on MIG devices is unsupported")
 	}
 
+	noSupportTimeSliceIDs := []string{}
+	for _, gpu := range devices.Gpu.Devices {
+		if !detectSupportTimeSliceByArch(gpu.architecture) {
+			klog.InfoS("the current card does not support setting time slices and will be ignored.", "arch", gpu.architecture, "uuid", gpu.uuid, "cudaComputeCapability", gpu.cudaComputeCapability)
+			noSupportTimeSliceIDs = append(noSupportTimeSliceIDs, gpu.uuid)
+			continue
+		}
+	}
+
+	supportTimeSliceIDs := difference(devices.UUIDs(), noSupportTimeSliceIDs)
+
+	if len(supportTimeSliceIDs) == 0 {
+		klog.InfoS("all card does not support setting time slices and will be ignored.", "uuids", noSupportTimeSliceIDs)
+	}
+
 	timeSlice := nascrd.DefaultTimeSlice
 	if config != nil && config.TimeSlice != nil {
 		timeSlice = *config.TimeSlice
 	}
 
-	err := t.nvdevlib.setComputeMode(devices.UUIDs(), "DEFAULT")
+	err := t.nvdevlib.setComputeMode(supportTimeSliceIDs, "DEFAULT")
```

Comment:
Note that `setComputeMode` is a different operation and should probably still be applied to **ALL** devices.

Reply:
done